Even more android fixes

matthewbauer opened the pull request:
###### Motivation for this change

This is a collection of tweaks for handling android. dc261f6896106deab2efd6329479d536d3e9c255 is the commit that solves my original problem & definitely should be merged if possible as it isn't a mass rebuild.

On the other hand, 62024d280eb7ecd1f49d80406193a65901317bcb, may not be that useful but it seems like a safe thing to just disable stripping & patchelf in cross drvs. Interested in feedback on whether this is a good idea.

It doesn't seem like 25e54c53a25f673f2ada852ae143f17dd1280842 should be needed but it fixed a non-fatal warning I was getting & might give us trouble if we don't properly set up the rpath.

matthewbauer committed:
libiconv: add "enableShared" option

When false, shared libraries will be disabled.

also adds patch to *really* disable shared libiconv.

GrahamcOfBorg commented:
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.42412&attempt_id=914daa46-d54c-4d30-bc05-a449ea9d84e1)

Attempted: libiconv

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/jcs8jiz94na13d960b8hd8b05j58i80x-glibc-iconv-2.27
```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.42412&attempt_id=bc566457-ee74-4227-b6c1-e64ee84ec1a1)

Attempted: libiconv

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/znlnr5p1n53n07d2mci10d2byl43gl2p-libiconv-osx-10.11.6
```
</p></details>

GrahamcOfBorg commented:
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.42412&attempt_id=35900b61-d9dc-4049-92c6-2de20d7f5fb5)

Attempted: libiconv

<details><summary>Partial log (click to expand)</summary><p>

```
  /nix/store/5fpq3zhq8dmvvc10q62i797j88l655y1-linux-headers-4.15
  /nix/store/csqjc2aycrin72amhc1xn49wwmcijddc-glibc-2.27-dev
  /nix/store/d5djbvna6ffvviqnkpy29p0798zprchr-glibc-2.27
  /nix/store/jcs8jiz94na13d960b8hd8b05j58i80x-glibc-iconv-2.27
copying path '/nix/store/d5djbvna6ffvviqnkpy29p0798zprchr-glibc-2.27' from 'https://cache.nixos.org'...
copying path '/nix/store/5fpq3zhq8dmvvc10q62i797j88l655y1-linux-headers-4.15' from 'https://cache.nixos.org'...
copying path '/nix/store/3nmzj02hcf38zzg1s6cl63g7n3801l5x-glibc-2.27-bin' from 'https://cache.nixos.org'...
copying path '/nix/store/csqjc2aycrin72amhc1xn49wwmcijddc-glibc-2.27-dev' from 'https://cache.nixos.org'...
copying path '/nix/store/jcs8jiz94na13d960b8hd8b05j58i80x-glibc-iconv-2.27' from 'https://cache.nixos.org'...
/nix/store/jcs8jiz94na13d960b8hd8b05j58i80x-glibc-iconv-2.27
```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.42412&attempt_id=af5635f0-18eb-4bcc-afd2-4b18de06c84b)

Attempted: libiconv

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/bbg8p1v9qhjfr0ghlzi8bn3f23wlj4rx-glibc-iconv-2.27
```
</p></details>

GrahamcOfBorg commented:
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.42412&attempt_id=4f334372-ce3b-447e-9eaf-b8fe3a28b227)

Attempted: libiconv

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/jcs8jiz94na13d960b8hd8b05j58i80x-glibc-iconv-2.27
```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.42412&attempt_id=d87cc4bd-513e-4bb4-bc09-e090ac6f4e3a)

Attempted: libiconv

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/znlnr5p1n53n07d2mci10d2byl43gl2p-libiconv-osx-10.11.6
```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.42412&attempt_id=b5ae325c-adfa-41c3-a6e2-7db1f2c0de20)

Attempted: libiconv

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/bbg8p1v9qhjfr0ghlzi8bn3f23wlj4rx-glibc-iconv-2.27
```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.42412&attempt_id=a6f1fc50-b1f3-42dc-95f5-b377e4c40774)

Attempted: libiconv

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/bbg8p1v9qhjfr0ghlzi8bn3f23wlj4rx-glibc-iconv-2.27
```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.42412&attempt_id=50435ce4-3b69-4819-86f5-c50239506fdb)

Attempted: libiconv

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/znlnr5p1n53n07d2mci10d2byl43gl2p-libiconv-osx-10.11.6
```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.42412&attempt_id=a7c0183b-d14b-451d-93c0-a9c9120b30c3)

Attempted: libiconv

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/bbg8p1v9qhjfr0ghlzi8bn3f23wlj4rx-glibc-iconv-2.27
```
</p></details>

GrahamcOfBorg commented:
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.42412&attempt_id=f029fe24-4c09-415e-80ac-bedc2a59adf5)

Attempted: libiconv

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/jcs8jiz94na13d960b8hd8b05j58i80x-glibc-iconv-2.27
```
</p></details>

matthewbauer commented on pkgs/development/mobile/androidenv/androidndk-pkgs.nix:
```diff
@@ -84,6 +84,7 @@ rec {
           $out/bin/${targetPlatform.config}-g++ \
           -e '130i    extraBefore+=(-Wl,--fix-cortex-a8)' \
           -e 's|^(extraBefore=)\(\)$|\1(${builtins.toString flags})|'
+        echo "-no-pie" >> $out/nix-support/cc-cflags
```
/cc @Ericson2314 Is this okay to do globally?

Ericson2314 replied:
Yeah sounds good. I'd maybe put it in the `extraBefore` thing above, but fine either way.

Ericson2314 commented on pkgs/os-specific/darwin/apple-source-releases/libiconv/default.nix:
```diff
@@ -1,15 +1,18 @@
-{ stdenv, appleDerivation, autoreconfHook, targetPlatform, enableStatic ? targetPlatform.isiOS }:
+{ stdenv, appleDerivation, lib, autoreconfHook, targetPlatform
+, enableStatic ? targetPlatform.isiOS
+, enableShared ? !targetPlatform.isiOS }:
```
can the `}:` be on its own line?

Ericson2314 commented on pkgs/os-specific/darwin/apple-source-releases/libiconv/default.nix:
```diff
@@ -1,15 +1,18 @@
   '';
 
-  configureFlags = stdenv.lib.optionals enableStatic [ "--enable-static" "--disable-shared" ];
+  configureFlags = lib.optional enableStatic "--enable-static"
```
let's use https://github.com/NixOS/nixpkgs/blob/master/lib/strings.nix#L415 on staging after?

matthewbauer replied:
Sounds good

Ericson2314 replied:
(And same for normal `libiconv`)

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.42412&attempt_id=129b2230-c2f2-4653-9758-79f9614a54e1)

Attempted: libiconv

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/nbiacn62k38saz9mb108b13gp9zcmrzc-glibc-iconv-2.27
```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.42412&attempt_id=dd3c902a-122f-44c5-b3a0-e31665f87228)

Attempted: libiconv

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/y0b2y33wzmlijsq81x831k3cy00v2vx4-libiconv-osx-10.11.6
```
</p></details>

GrahamcOfBorg commented:
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.42412&attempt_id=c5f58f27-5c4f-4e98-a046-1ba3ad57bce1)

Attempted: libiconv

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/yhv02gycgbl4zrvpa8kjgxl7b1wh8c2x-glibc-iconv-2.27
```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.42412&attempt_id=879cd899-1913-4b8c-a0f9-16558daea52b)

Attempted: libiconv

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/nbiacn62k38saz9mb108b13gp9zcmrzc-glibc-iconv-2.27
```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.42412&attempt_id=9a8d7e16-bbb6-4d56-83f1-e87d7cb5f5d1)

Attempted: libiconv

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/y0b2y33wzmlijsq81x831k3cy00v2vx4-libiconv-osx-10.11.6
```
</p></details>

GrahamcOfBorg commented:
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.42412&attempt_id=a6ced023-4fe4-40f9-9831-e4da34f0ddfe)

Attempted: libiconv

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/yhv02gycgbl4zrvpa8kjgxl7b1wh8c2x-glibc-iconv-2.27
```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.42412&attempt_id=d522c83d-1443-4420-bf60-6424b337f5bd)

Attempted: libiconv

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/y0b2y33wzmlijsq81x831k3cy00v2vx4-libiconv-osx-10.11.6
```
</p></details>

GrahamcOfBorg commented:
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.42412&attempt_id=8b9553ec-455a-4a37-8a21-7e8ec8793dd0)

Attempted: libiconv

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/yhv02gycgbl4zrvpa8kjgxl7b1wh8c2x-glibc-iconv-2.27
```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.42412&attempt_id=678dc88b-ff17-4f70-a8e3-d33ca10a7e88)

Attempted: libiconv

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/nbiacn62k38saz9mb108b13gp9zcmrzc-glibc-iconv-2.27
```
</p></details>

matthewbauer committed:
android: add --build-id to ldflags

the gcc in android does not have this on by default. Doing this will
fix some of the errors in cross-trunk:

https://hydra.nixos.org/build/76354538/nixlog/11

This is related to how separate debug info is handled in Nixpkgs.

matthewbauer committed:
androidndk: add fullNDK option

Setting this to true will disable removing some of the unused parts of
the NDK toolchain that is unused in Nixpkgs but may be used by others.

/cc @bkchr

matthewbauer committed:
ghc: build shared on android

We run into issues otherwise

GrahamcOfBorg commented:
Success on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.42412&attempt_id=15d04087-58ad-4708-ac4c-a3888cbdc0d3)

Attempted: libiconv

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/y0b2y33wzmlijsq81x831k3cy00v2vx4-libiconv-osx-10.11.6
```
</p></details>

GrahamcOfBorg commented:
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.42412&attempt_id=3285672c-dffe-4d47-aa81-d51406e91897)

Attempted: libiconv

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/yhv02gycgbl4zrvpa8kjgxl7b1wh8c2x-glibc-iconv-2.27
```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.42412&attempt_id=63f1f492-c414-4387-8050-72e810f906eb)

Attempted: libiconv

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/nbiacn62k38saz9mb108b13gp9zcmrzc-glibc-iconv-2.27
```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.42412&attempt_id=21ee21c6-8349-4bac-b7e5-18671d9764ee)

Attempted: libiconv

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/y0b2y33wzmlijsq81x831k3cy00v2vx4-libiconv-osx-10.11.6
```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.42412&attempt_id=0197c3bf-ec0a-4cbb-a3af-a078a3cfb394)

Attempted: libiconv

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/nbiacn62k38saz9mb108b13gp9zcmrzc-glibc-iconv-2.27
```
</p></details>

GrahamcOfBorg commented:
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.42412&attempt_id=062ee407-c420-4ba4-94dc-7731f9b524f2)

Attempted: libiconv

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/yhv02gycgbl4zrvpa8kjgxl7b1wh8c2x-glibc-iconv-2.27
```
</p></details>

GrahamcOfBorg commented:
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.42412&attempt_id=0adcf8fb-e696-44a1-810e-3e7663172909)

Attempted: libiconv

The following builds were skipped because they don't evaluate on aarch64-linux: ghc

<details><summary>Partial log (click to expand)</summary><p>

```
a) For `nixos-rebuild` you can set
  { nixpkgs.config.allowUnsupportedSystem = true; }
in configuration.nix to override this.

b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowUnsupportedSystem = true; }
to ~/.config/nixpkgs/config.nix.


/nix/store/yhv02gycgbl4zrvpa8kjgxl7b1wh8c2x-glibc-iconv-2.27
```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.42412&attempt_id=f85599a3-bd89-4be2-95e3-4054faf05b43)

Attempted: ghc, libiconv

<details><summary>Partial log (click to expand)</summary><p>

```
these paths will be fetched (88.74 MiB download, 900.17 MiB unpacked):
  /nix/store/bdvs7h7zirrfjrk7nn2cm2w8cnivyvdh-ghc-8.2.2
copying path '/nix/store/bdvs7h7zirrfjrk7nn2cm2w8cnivyvdh-ghc-8.2.2' from 'https://cache.nixos.org'...
/nix/store/bdvs7h7zirrfjrk7nn2cm2w8cnivyvdh-ghc-8.2.2
/nix/store/y0b2y33wzmlijsq81x831k3cy00v2vx4-libiconv-osx-10.11.6
```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.42412&attempt_id=347e889b-741b-40ff-8559-4e764373bc54)

Attempted: ghc, libiconv

<details><summary>Partial log (click to expand)</summary><p>

```
these paths will be fetched (100.89 MiB download, 1306.19 MiB unpacked):
  /nix/store/i2d6l6w99qlwhz6gbz4hxl4d9238drfj-ghc-8.2.2
  /nix/store/yzphq2ly3r11jwai4ya7snnl6p1vd7kd-ghc-8.2.2-doc
copying path '/nix/store/yzphq2ly3r11jwai4ya7snnl6p1vd7kd-ghc-8.2.2-doc' from 'https://cache.nixos.org'...
copying path '/nix/store/i2d6l6w99qlwhz6gbz4hxl4d9238drfj-ghc-8.2.2' from 'https://cache.nixos.org'...
/nix/store/i2d6l6w99qlwhz6gbz4hxl4d9238drfj-ghc-8.2.2
/nix/store/nbiacn62k38saz9mb108b13gp9zcmrzc-glibc-iconv-2.27
```
</p></details>

matthewbauer committed:
android: define __ANDROID_API__ in cflags

This is needed for targeting older apis.

GrahamcOfBorg commented:
Success on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.42412&attempt_id=d29b9d84-16e3-4694-87c2-4b9856195063)

Attempted: ghc, libiconv

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/bdvs7h7zirrfjrk7nn2cm2w8cnivyvdh-ghc-8.2.2
/nix/store/y0b2y33wzmlijsq81x831k3cy00v2vx4-libiconv-osx-10.11.6
```
</p></details>

GrahamcOfBorg commented:
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.42412&attempt_id=17d73633-e985-4c66-a602-f9a507e191ea)

Attempted: libiconv

The following builds were skipped because they don't evaluate on aarch64-linux: ghc

<details><summary>Partial log (click to expand)</summary><p>

```
a) For `nixos-rebuild` you can set
  { nixpkgs.config.allowUnsupportedSystem = true; }
in configuration.nix to override this.

b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowUnsupportedSystem = true; }
to ~/.config/nixpkgs/config.nix.


/nix/store/yhv02gycgbl4zrvpa8kjgxl7b1wh8c2x-glibc-iconv-2.27
```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.42412&attempt_id=ed6f8ef7-d9c5-44d9-aad9-0b70fcf6ae5f)

Attempted: ghc, libiconv

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/i2d6l6w99qlwhz6gbz4hxl4d9238drfj-ghc-8.2.2
/nix/store/nbiacn62k38saz9mb108b13gp9zcmrzc-glibc-iconv-2.27
```
</p></details>

matthewbauer committed:
impure.nix: fix handling of localSystem

If we passed a localSystem in, we don’t want the current system to
override it. Now we check for localSystem first to avoid getting
"mixed" localSystem values from commands like this:

  nix-build --arg localSystem '{config="x86_64-unknown-linux-musl";}' -A hello

Which would eventually evaluate localSystem to this:

{
  config = "x86_64-unknown-linux-musl";
  system = "x86_64-darwin";
}

& Nix would not be able to run it correctly.

Fixes #41599

/cc @Ericson2314

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.42412&attempt_id=ed761863-22c7-4d7b-aa75-bd586e51d668)

Attempted: ghc, libiconv

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/i2d6l6w99qlwhz6gbz4hxl4d9238drfj-ghc-8.2.2
/nix/store/nbiacn62k38saz9mb108b13gp9zcmrzc-glibc-iconv-2.27
```
</p></details>

GrahamcOfBorg commented:
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.42412&attempt_id=7150e999-be96-41de-a6e9-3c642e807919)

Attempted: libiconv

The following builds were skipped because they don't evaluate on aarch64-linux: ghc

<details><summary>Partial log (click to expand)</summary><p>

```

b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowUnsupportedSystem = true; }
to ~/.config/nixpkgs/config.nix.


these paths will be fetched (0.00 MiB download, 0.00 MiB unpacked):
  /nix/store/yhv02gycgbl4zrvpa8kjgxl7b1wh8c2x-glibc-iconv-2.27
copying path '/nix/store/yhv02gycgbl4zrvpa8kjgxl7b1wh8c2x-glibc-iconv-2.27' from 'https://cache.nixos.org'...
/nix/store/yhv02gycgbl4zrvpa8kjgxl7b1wh8c2x-glibc-iconv-2.27
```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.42412&attempt_id=4ad058ae-3a0a-4ae6-99f1-63c9e8fbddc8)

Attempted: ghc, libiconv

<details><summary>Partial log (click to expand)</summary><p>

```
these paths will be fetched (88.74 MiB download, 900.17 MiB unpacked):
  /nix/store/bdvs7h7zirrfjrk7nn2cm2w8cnivyvdh-ghc-8.2.2
copying path '/nix/store/bdvs7h7zirrfjrk7nn2cm2w8cnivyvdh-ghc-8.2.2' from 'https://cache.nixos.org'...
/nix/store/bdvs7h7zirrfjrk7nn2cm2w8cnivyvdh-ghc-8.2.2
/nix/store/y0b2y33wzmlijsq81x831k3cy00v2vx4-libiconv-osx-10.11.6
```
</p></details>

matthewbauer committed:
ghc843: don’t apply stripping in android

This breaks some library symbols apparently. ARM symbols must be
incorrectly stripped. Also some very weird issues with patchelf.

GrahamcOfBorg commented:
Success on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.42412&attempt_id=abc7e1ba-8dd4-4529-a861-bda2489694c0)

Attempted: ghc, libiconv

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/bdvs7h7zirrfjrk7nn2cm2w8cnivyvdh-ghc-8.2.2
/nix/store/y0b2y33wzmlijsq81x831k3cy00v2vx4-libiconv-osx-10.11.6
```
</p></details>

GrahamcOfBorg commented:
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.42412&attempt_id=8fef9123-af1a-4067-83cf-eecd77dae18d)

Attempted: libiconv

The following builds were skipped because they don't evaluate on aarch64-linux: ghc

<details><summary>Partial log (click to expand)</summary><p>

```
a) For `nixos-rebuild` you can set
  { nixpkgs.config.allowUnsupportedSystem = true; }
in configuration.nix to override this.

b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowUnsupportedSystem = true; }
to ~/.config/nixpkgs/config.nix.


/nix/store/yhv02gycgbl4zrvpa8kjgxl7b1wh8c2x-glibc-iconv-2.27
```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.42412&attempt_id=f834e0f4-078a-40b5-9022-c23cbc27fd53)

Attempted: ghc, libiconv

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/i2d6l6w99qlwhz6gbz4hxl4d9238drfj-ghc-8.2.2
/nix/store/nbiacn62k38saz9mb108b13gp9zcmrzc-glibc-iconv-2.27
```
</p></details>

Ericson2314 commented on pkgs/top-level/impure.nix:
```diff
@@ -83,5 +83,6 @@ import ./. (builtins.removeAttrs args [ "system" "platform" ] // {
   # Autotools parlance) system.
-  localSystem = { system = builtins.currentSystem; } // localSystem;
+  localSystem = (if args ? localSystem then {}
+                 else { system = builtins.currentSystem; }) // localSystem;
```
Ah, nice catch.

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.42412&attempt_id=32a68a41-3280-49b5-8413-140364bd98e6)

Attempted: ghc, libiconv

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/i2d6l6w99qlwhz6gbz4hxl4d9238drfj-ghc-8.2.2
/nix/store/nbiacn62k38saz9mb108b13gp9zcmrzc-glibc-iconv-2.27
```
</p></details>

GrahamcOfBorg commented:
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.42412&attempt_id=3a5ee45b-5b08-4f61-a269-a3c0168c1668)

Attempted: libiconv

The following builds were skipped because they don't evaluate on aarch64-linux: ghc

<details><summary>Partial log (click to expand)</summary><p>

```
a) For `nixos-rebuild` you can set
  { nixpkgs.config.allowUnsupportedSystem = true; }
in configuration.nix to override this.

b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowUnsupportedSystem = true; }
to ~/.config/nixpkgs/config.nix.


/nix/store/yhv02gycgbl4zrvpa8kjgxl7b1wh8c2x-glibc-iconv-2.27
```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.42412&attempt_id=817e212c-e1fa-45b8-b773-59477efbf134)

Attempted: ghc, libiconv

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/i2d6l6w99qlwhz6gbz4hxl4d9238drfj-ghc-8.2.2
/nix/store/nbiacn62k38saz9mb108b13gp9zcmrzc-glibc-iconv-2.27
```
</p></details>

GrahamcOfBorg commented:
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.42412&attempt_id=d964c231-4779-4651-bc7b-5be57c57e66c)

Attempted: libiconv

The following builds were skipped because they don't evaluate on aarch64-linux: ghc

<details><summary>Partial log (click to expand)</summary><p>

```
a) For `nixos-rebuild` you can set
  { nixpkgs.config.allowUnsupportedSystem = true; }
in configuration.nix to override this.

b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowUnsupportedSystem = true; }
to ~/.config/nixpkgs/config.nix.


/nix/store/yhv02gycgbl4zrvpa8kjgxl7b1wh8c2x-glibc-iconv-2.27
```
</p></details>

GrahamcOfBorg commented:
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.42412&attempt_id=31335f01-1fea-4834-80ff-853ac57afa12)

Attempted: libiconv

The following builds were skipped because they don't evaluate on aarch64-linux: ghc

<details><summary>Partial log (click to expand)</summary><p>

```
a) For `nixos-rebuild` you can set
  { nixpkgs.config.allowUnsupportedSystem = true; }
in configuration.nix to override this.

b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowUnsupportedSystem = true; }
to ~/.config/nixpkgs/config.nix.


/nix/store/yhv02gycgbl4zrvpa8kjgxl7b1wh8c2x-glibc-iconv-2.27
```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.42412&attempt_id=21d9d607-fd50-4ffb-96fb-605651dba8d9)

Attempted: ghc, libiconv

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/i2d6l6w99qlwhz6gbz4hxl4d9238drfj-ghc-8.2.2
/nix/store/nbiacn62k38saz9mb108b13gp9zcmrzc-glibc-iconv-2.27
```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.42412&attempt_id=c6b11a05-3766-4485-b298-70c82e9e8cdf)

Attempted: ghc, libiconv

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/bdvs7h7zirrfjrk7nn2cm2w8cnivyvdh-ghc-8.2.2
/nix/store/y0b2y33wzmlijsq81x831k3cy00v2vx4-libiconv-osx-10.11.6
```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.42412&attempt_id=01e318ad-7573-46bf-949b-63cd69b80b32)

Attempted: ghc, libiconv

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/bdvs7h7zirrfjrk7nn2cm2w8cnivyvdh-ghc-8.2.2
/nix/store/y0b2y33wzmlijsq81x831k3cy00v2vx4-libiconv-osx-10.11.6
```
</p></details>

GrahamcOfBorg commented:
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.42412&attempt_id=a0595cd4-2cc2-4116-a98c-92b06c80e96d)

Attempted: libiconv

The following builds were skipped because they don't evaluate on aarch64-linux: ghc

<details><summary>Partial log (click to expand)</summary><p>

```
a) For `nixos-rebuild` you can set
  { nixpkgs.config.allowUnsupportedSystem = true; }
in configuration.nix to override this.

b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowUnsupportedSystem = true; }
to ~/.config/nixpkgs/config.nix.


/nix/store/yhv02gycgbl4zrvpa8kjgxl7b1wh8c2x-glibc-iconv-2.27
```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.42412&attempt_id=7249ea9d-07e3-45b6-b11a-23152cfb67a8)

Attempted: ghc, libiconv

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/i2d6l6w99qlwhz6gbz4hxl4d9238drfj-ghc-8.2.2
/nix/store/nbiacn62k38saz9mb108b13gp9zcmrzc-glibc-iconv-2.27
```
</p></details>

layus commented:
Thanks! he "impure.nix: fix handling of localSystem" explains why it was such a pain for me to pass that argument.
